File: CSharp/LeetCode/Solutions/1957-delete-characters-to-make-fancy-string.cs
using LeetCode.Solutions.Shared;
using System.Text;

namespace LeetCode.Solutions;

internal class Solution1957 : IRunProgram
{
    public void Run()
    {

    }

    public string MakeFancyString(string s)
    {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < s.Length; i++)
        {
            if (sb.Length < 2 || (sb[sb.Length - 2] != sb[sb.Length - 1] || sb[sb.Length - 1] != s[i]))
            {
                sb.Append(s[i]);
            }
        }
        return sb.ToString();
    }
}

File: Java/1957-delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder sb = new StringBuilder();

        for (char ch : s.toCharArray()) {
            if (sb.length() < 2 || sb.charAt(sb.length() - 2) != sb.charAt(sb.length() - 1) ||  sb.charAt(sb.length() - 1) != ch) {
                sb.append(ch);
            }
        }

        return sb.toString();
    }
}